action/output.php

```diff
@@ -54,11 +38,11 @@ public function handle_read(Doku_Event $event, $param) {
      * @param $param
      */
     public function handle_output(Doku_Event $event, $param) {
-        global $ID;
-        if($this->lastread != $ID) return; // avoid nested calls
-        $this->lastread = '';
+        global $ID, $ACT;
+        if (act_clean($ACT) !== 'show') {
+            return;
```

Comment:
Do we? Under what circumstances would a page containing struct data be rendered in a mode where we do not want the struct output?